timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -164,14 +211,14 @@ void __global__ k_inv_permute_accum(
         return;
     }
 
-    array[perm[idx] * stride + stride_idx] += sorted_array[idx * stride + stride_idx];
+    array[idxs[idx] * stride + stride_idx] += gathered_array[idx * stride + stride_idx];
```

Comment:
A user of this kernel could accidentally cause a race condition by including repeated elements in `idxs` (previously `idxs` was a permutation, but now `idxs` could be more general).

Could we add a line comment that `idxs` is assumed to contain unique indices, and explicitly assert uniqueness near where `idxs` is defined?

Reply:
@maxentile Good call. I renamed `idxs` -> `unique_idxs` and added line comments in 77508be.

> explicitly assert uniqueness near where idxs is defined?

Hmm, I think it is a bit tricky to do this right at the call site, since `d_atom_idxs` is already in device memory. We do already require `atom_idxs` to be a set in the [constructor argument](https://github.com/proteneer/timemachine/blob/77508bef811eb25a529a758b16d3a1f005b09411/timemachine/cpp/src/nonbonded_all_pairs.cu#L32), so this may be sufficient? What do you think?

Reply:
Looks good!

>Hmm, I think it is a bit tricky to do this right at the call site, since d_atom_idxs is already in device memory.

Ahh, good point

>We do already require atom_idxs to be a set in the [constructor argument](https://github.com/proteneer/timemachine/blob/77508bef811eb25a529a758b16d3a1f005b09411/timemachine/cpp/src/nonbonded_all_pairs.cu#L32), so this may be sufficient? What do you think?

I think that is sufficient!